### ISSUE TITLE
Issue in training a new language 

### ISSUE THREAD

**Issue by Ham714** [PaddlePaddle/PaddleOCR#8906]
I am trying to train a model for Greek + English language using svtrnet. My model is getting trained and its accuracy is increasing but when i tried to infer it it shows an error also shows results nut is was very poor. kindly someone help me.

this is my train log 
[train.log](https://github.com/PaddlePaddle/PaddleOCR/files/10508683/train.log)
this is the error that i am getting 
![image](https://user-images.githubusercontent.com/119650336/214823284-3ca96356-c936-4630-a0c4-67b7450a7b2e.png)
this is my config file that i am using 
[config.txt](https://github.com/PaddlePaddle/PaddleOCR/files/10508703/config.txt)

kindly someone help me plz @WenmuZhou  @dyning  @LDOUBLEV  @tink2123  @MissPenguin  @Evezerest  @littletomatodonkey 
@andyjpaddle   @D-DanielYang  @Topdu  @sdcb  @ZeyuChen  @bingooo

i can also upload the dataset if someone want. I used SynthTIGER to generate 10000 images for training and 1000 for validating

plz guide me

**Comment by Ham714** [comment]
i dont have a GPU based hardware that why im trying to train it on CPU @WenmuZhou Guide me plz

**Comment by Ham714** [comment]
oky now i have overcome the error now the results is very bad not even close to it 
DO I HAVE TO WAIT FOR THE TRAINING TO COMPLETE

**Comment by Ham714** [comment]
Anyone  @WenmuZhou @dyning @LDOUBLEV @tink2123 @MissPenguin @Evezerest @littletomatodonkey
@andyjpaddle @D-DanielYang @Topdu @sdcb @ZeyuChen @bingooo

**Comment by Ham714** [comment]
I also export the infer model to check it bt python script but i giving empty brackets
![image](https://user-images.githubusercontent.com/119650336/215026149-c3a2573d-4130-43b9-99d1-d18925e94733.png)
![image](https://user-images.githubusercontent.com/119650336/215026278-13eebd3b-4602-4787-94bf-ff77d107802c.png)
can anyone mention the mistake i am doing

**Comment by Ham714** [comment]
@WenmuZhou plz

**Comment by Topdu** [comment]
Trying to add paramters plz: rec_algorithm='SVTR', rec_image_shape='3,64,256'.

**Comment by Ham714** [comment]
> Trying to add paramters plz: rec_algorithm='SVTR', rec_image_shape='3,64,256'.

@Topdu Thank you soo much for the reply i have done this as you ask but now this error is pop up
![image](https://user-images.githubusercontent.com/119650336/215288000-111316f0-d5a6-4096-be80-e29216d87cbc.png)

**Comment by Topdu** [comment]
refer to [predict system](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/inference_ppocr_en.md#text-detection-angle-classification-and-recognition-inference-concatenation)

**Comment by Ham714** [comment]
> refer to [predict system](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/inference_ppocr_en.md#text-detection-angle-classification-and-recognition-inference-concatenation)

Thank You @Topdu But im not getting anything, i even tried to modified paddleocr.py manually but still empty bracket, plz help me

**Comment by Ham714** [comment]
@Topdu did I make a mistake in training 
[train.log](https://github.com/PaddlePaddle/PaddleOCR/files/10532931/train.log)

**Comment by Topdu** [comment]
python tools/infer/predict_system.py --image_dir="./your_img_path" --det_model_dir="./ch_PP-OCRv3_det_infer/" --rec_model_dir="./svtr_rec/" --rec_char_dict_path='greek_dict.txt' --use_angle_cls=false --rec_algorithm='SVTR' --rec_image_shape='3,64,256' --use_sapce_char=False

**Comment by Ham714** [comment]
I tried on another image, this is i get
![image](https://user-images.githubusercontent.com/119650336/215403818-06e99ee9-8555-4ef4-96b1-5382b9972cae.png)
![image](https://user-images.githubusercontent.com/119650336/215403754-0ac46991-8977-431b-8c93-31f4eacf7e5a.png)
this is the result i get when i tried on validation image
![image](https://user-images.githubusercontent.com/119650336/215405196-2e496dbb-cc08-4f61-97a3-d053fae89db0.png)
@Topdu

**Comment by Topdu** [comment]
If allowed, upload your trained model and test images please. I'll try it later.

**Comment by Ham714** [comment]
[inference.zip](https://github.com/PaddlePaddle/PaddleOCR/files/10533279/inference.zip)
[svtrnet.zip](https://github.com/PaddlePaddle/PaddleOCR/files/10533280/svtrnet.zip)
do you also need dataset?
BTW these are the type of images i want to extract text. thanks if you will help me

**Comment by Ham714** [comment]
> [inference.zip](https://github.com/PaddlePaddle/PaddleOCR/files/10533279/inference.zip) [svtrnet.zip](https://github.com/PaddlePaddle/PaddleOCR/files/10533280/svtrnet.zip) do you also need dataset? BTW these are the type of images i want to extract text. thanks if you will help me
@Topdu

**Comment by Ham714** [comment]
[config.txt](https://github.com/PaddlePaddle/PaddleOCR/files/10534496/config.txt)
@Topdu i wana ask if there is any problem in my config file cause i have cpu based system and my training takes 3 days to complete. Well right now im only using small dataset(train 10000 val 1000) to see if its gives results. Once i got some results, i will get the gpu based system and try to train on bigger dataset for the dataset i also upload for you to see if there is any problem in dataset
you can download the dataset using this link https://file.io/USrwQLHWDZAM

**Comment by Topdu** [comment]
there is no problem in config file. But it would be helpful if you could provide us with details of how config file was modified.

**Comment by Ham714** [comment]
> there is no problem in config file. But it would be helpful if you could provide us with details of how config file was modified.

@Topdu i already send my **config file** in above comment and also the **train.log**

**Comment by Ham714** [comment]
@Topdu did you check my model? kindly take a look at that also i have share the dataset if you wana take a look

**Comment by Topdu** [comment]
upload the char dictionary file please.

**Comment by Ham714** [comment]
[greek_dict.txt](https://github.com/PaddlePaddle/PaddleOCR/files/10542278/greek_dict.txt)
Sorry for the late reply @Topdu

**Comment by Topdu** [comment]
upload the trained model instead of inference model plz.

**Comment by Topdu** [comment]
max_text_length should be set to 25, and use as large batchsize and as much training data as possible. Adjust the learning rate while adjusting the batchsize refer to[5.5](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/algorithm_rec_svtr_en.md#5-faq)

**Comment by Ham714** [comment]
@Topdu here is the trained model https://file.io/3UKe6OUUsHzd

**Comment by Topdu** [comment]
![image](https://user-images.githubusercontent.com/28981817/215669059-6023f0d7-dd98-45da-9dcc-1f002f80239f.png)
![image](https://user-images.githubusercontent.com/28981817/215669133-b523f807-b9f8-47e8-bebe-a0110c1e23f1.png)
Why are there duplicate letters in the dict dictionary file?

**Comment by Ham714** [comment]
> ![image](https://user-images.githubusercontent.com/28981817/215669059-6023f0d7-dd98-45da-9dcc-1f002f80239f.png) ![image](https://user-images.githubusercontent.com/28981817/215669133-b523f807-b9f8-47e8-bebe-a0110c1e23f1.png) Why are there duplicate letters in the dict dictionary file?

@Topdu i just merge greek and English dict alphabets together. So is that the problem ?

**Comment by Ham714** [comment]
> max_text_length should be set to 25, and use as large batchsize and as much training data as possible. Adjust the learning rate while adjusting the batchsize refer to[5.5](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/algorithm_rec_svtr_en.md#5-faq). You can also use svtr's [English model](https://paddleocr.bj.bcebos.com/PP-OCRv3/chinese/rec_svtr_tiny_none_ctc_en_train.tar) as a pre-trained model to speed up training.

![image](https://user-images.githubusercontent.com/119650336/215671280-539662bc-4f5e-4f82-a723-bed30064f979.png)
is this possible cause the finetuning documentation say that it is not recommended if dict is changed

**Comment by Ham714** [comment]
> > max_text_length should be set to 25, and use as large batchsize and as much training data as possible. Adjust the learning rate while adjusting the batchsize refer to[5.5](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/doc/doc_en/algorithm_rec_svtr_en.md#5-faq). You can also use svtr's [English model](https://paddleocr.bj.bcebos.com/PP-OCRv3/chinese/rec_svtr_tiny_none_ctc_en_train.tar) as a pre-trained model to speed up training.
> 
> ![image](https://user-images.githubusercontent.com/119650336/215671280-539662bc-4f5e-4f82-a723-bed30064f979.png) 
And Also @Topdu is this possible cause the finetuning documentation say that it is not recommended if dict is changed

**Comment by Topdu** [comment]
> i just merge greek and English dict alphabets together. So is that the problem ?

The characters in the dictionary file should be unique.


> And Also @Topdu is this possible cause the finetuning documentation say that it is not recommended if dict is changed

It means that if the dictionary is changed, more training data is needed.

**Comment by Ham714** [comment]
> > i just merge greek and English dict alphabets together. So is that the problem ?
> 
> The characters in the dictionary file should be unique.
> 
> > And Also @Topdu is this possible cause the finetuning documentation say that it is not recommended if dict is changed
> 
> It means that if the dictionary is changed, more training data is needed.

well this means i have to train model again and wait for 2 more days thank you @Topdu  i will let you know once i finish my training again

**Comment by Ham714** [comment]
![image](https://user-images.githubusercontent.com/119650336/215679963-e534da69-7829-4ac5-9f61-3e78367ff797.png)

But @Topdu  why did it not gives output and apart from this in the above figure it already show that it detect 35 and recognize 35 box

**Comment by jaihuayen** [comment]
@Ham714 You need to adjust the drop_score to a lower level so as to show the lower confidence recognition results

**Comment by rm-asif-amin** [comment]
@Topdu 
Hi. Could you have a look and share your opinion? I'm trying to train for bengali(it's not supported by paddle yet). Here's my approach and outcome-
1. Training a model from scratch with the following yml config(click to expand) -
<details>
Global:
  use_gpu: true
  epoch_num: 10
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_ben
  save_epoch_step: 3
  eval_batch_step:
  - 0
  - 100
  cal_metric_during_train: true
  pretrained_model: null
  checkpoints: null
  save_inference_dir: /content/inference
  use_visualdl: false
  infer_img: 
  character_dict_path: /content/bn_dict.txt
  max_text_length: 25
  infer_mode: false
  use_space_char: true
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
  regularizer:
    name: L2
    factor: 1.0e-05
Architecture:
  model_type: rec
  algorithm: CRNN
  Transform: null
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: small
    small_stride:
    - 1
    - 2
    - 2
    - 2
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 48
  Head:
    name: CTCHead
    fc_decay: 1.0e-05
Loss:
  name: CTCLoss
PostProcess:
  name: CTCLabelDecode
Metric:
  name: RecMetric
  main_indicator: acc
Train:
  dataset:
    name: SimpleDataSet
    data_dir: /content/content/final_dataset/rec/train/train
    label_file_list:
    - /content/content/final_dataset/rec/train/rec_gt_train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug: null
    - CTCLabelEncode: null
    - RecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: true
    batch_size_per_card: 256
    drop_last: true
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /content/content/final_dataset/rec/test/test
    label_file_list:
    - /content/content/final_dataset/rec/test/rec_gt_test.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - CTCLabelEncode: null
    - RecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 256
    num_workers: 8

</details>
2. Using a dataset of ~80k one word images. After training for 100+ epochs, the training loss looks like this-

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/cbca8b51-f048-4fef-a0cd-f15ebff24ca0)
4. The training, eval accuracy are not coherent ( low eval accuracy compared to train)-
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/e0a62f6c-c922-48a3-9f1d-cabc7ff26001)

5. The Edit distance metrics, on the other hand are opposite -
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/b3abfaf6-a2e7-4b4b-9dfa-aa9ff84ba2a0)
6. I've added Bengali letters to an English dictionary which seems to be encoding correctly-
```
Original Label: ব্যঙ্গবিদ্রূপাত্মক
Transformed Label: [157 160 139 137 157 123 152 161 126 155 122 150 159 135   0   0   0   0
   0   0   0   0   0   0   0]
```
My questions are-
1. Is my approach right? The model seems to be learning, why the low accuracy?
2. Should I use a pre-trained model and finetune?( A similar indic language maybe? )

Would love to know your opinion.
